### PR TITLE
Foreign Keys with cascading deletes

### PR DESCRIPTION
**Pull request by CodingGorilla**
Adding the ability to cascade deletes when creating a foreign key

### CODE REVIEW

File: src/Marten.Testing/Examples/ForeignKeyExamples.cs [hunk 1]
```diff
@@ -20,6 +20,32 @@ public void configuration()
 
             //var sql = store.Schema.ToDDL();
             //Console.WriteLine(sql);
+        }
+
+        public void external_fkey()
+        {
+            // SAMPLE: configure-external-foreign-key
+var store = DocumentStore.For(_ =>
```
Comment (jeremydmiller) [inline src/Marten.Testing/Examples/ForeignKeyExamples.cs]: You don't have to left justify code samples anymore. I fixed that in stdocs quite awhile back.

File: src/Marten.Testing/Acceptance/foreign_keys.cs [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+using Marten.Testing.Documents;
+using Shouldly;
+using Xunit;
+
+namespace Marten.Testing.Acceptance
+{
+    public class foreign_keys: IntegratedFixture
+    {
+        [Fact]
+        public void can_insert_document_with_null_value_of_foregin_key()
```
Comment (mysticmind) [inline src/Marten.Testing/Acceptance/foreign_keys.cs]: @oskardudycz Please fix typo in the method name `foregin` to `foreign`? Looks like this it is there in all the unit test method names.

Comment (oskardudycz) [reply src/Marten.Testing/Acceptance/foreign_keys.cs]: @mysticmind fixed. There is nothing better than copy/pasting mistakes... 😄

**Comment by mysticmind** [comment]
With regards to failing build, it is due to a random blinking test pertaining to `Marten.Testing.Linq.query_with_dates_Tests.can_select_DateTimeOffset_and_will_return_localtime`

**Comment by oskardudycz** [comment]
@mysticmind as discussed, I'll try to tackle that in separate PR, as this blinker is not related to this PR.

**Comment by CodingGorilla** [comment]
@oskardudycz Looks good 👍   Thanks again for finishing this up for me.

**Pull request merged**